### DISCUSSION TITLE
NVDA GPU Driect RDMA support in Windows with Libfabric

### DISCUSSION THREAD

**Discussion by sen-sd** [ofiwg/libfabric#8342] [Q&A] [answered]
Is libfabric support GPU Driect RDMA. Based on code I found some information like it support for linux.
prov\psm3\psm3\psm_rndv_mod.c
static uint64_t get_nvidia_bar_addr(int domain, int bus, int slot) -> in this function it is reading bus address with Linux API call.
Please confirm GPU Driect RDMA support for Windows is available or not.

**Comment by shefty** [comment, marked as answer]
No, AFAIK, the Windows OS does not support the necessary kernel mechanism required to setup PCI peer-to-peer device transfers, or expose such a mechanism to user-space.

The Linux OS mechanism is known as dma-buf.
